Added adoption script docs (#1103)
demisto/content-docs: docs/partners/adopt.md

![pack example cyren](/doc_imgs/partners/packexample_cyren.png)

## Process
Choose one of the options below and follow the step by step instructions or watch the video tutorials below to learn how to submit your adoption Pull Requests. 

<details><summary><strong>Working on a local clone of Content repo</strong></summary>
<div>

<br/>
These methods assume that you have already [forked the `content` repository](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-2-fork-the-github-repo) and [cloned the fork onto your local machine](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-3-clone-the-github-fork-locally).
Choose either one of the methods below to start or complete adoption.

<br/>
<details><summary><strong>Adopt Using Helper Script</strong></summary>

<br/>

<div>This script will automatically perform the necessary steps to create an adoption PR.

> **Note:** The script was only tested on Unix systems, specifically Mac OS and Ubuntu. It might not work correctly on other systems. 

Reviewer: I don't relate to this disclaimer as it doesn't give anything to the user besides the feeling we are not taking responsibility if this fails. Instead, I think we should mention that the script is supported in Linux and Mac OS and if there's an issue the user should consult his Rep. from the PS team.

**Requirements:**
Before using this automation, make sure you have [`git`](https://git-scm.com/downloads) and [`python3`](https://www.python.org/downloads/) installed and in your `PATH`. This script will install [`demisto-sdk`](https://github.com/demisto/demisto-sdk#installation) python package if it does not exist in your environment.

Follow the steps below to adopt using the helper script:

> 1. Change directory to where the `content` repository is located. This is the location where you [cloned the forked `content` repository](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-3-clone-the-github-fork-locally).

Reviewer: ```suggestion
1. Inside your terminal, change your working directory to the root of the Content repository. This is the location where you [cloned the forked `content` repository](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-3-clone-the-github-fork-locally).
```

> 2. Look for the Pack you want to adopt in the `content/Packs/` directory. You will use the folder name as the second argument to the `adopt_pack.bash` script. 

Reviewer: They should already be in content folder
```suggestion
2. Look for the pack you want to adopt under the `Packs/` directory. You will use the folder name as the second argument to the `adopt_pack.bash` script. 
```

For example, if we wanted to start adopting the `HelloWorld` Pack, we would run the following command:

```bash
PACK=HelloWorld
./Utils/adopt_pack.bash start $PACK
```

3. **After 90 days**, run the script below to complete the adoption:

```bash
PACK=HelloWorld
./Utils/adopt_pack.bash complete $PACK
```

You will be prompted for the following information when running this command:

- Your organization/company's name.
- A link to your organization's support site.
- Email address for your organization's support.
- A link to download your [author image](https://xsoar.pan.dev/docs/packs/packs-format#author_imagepng). If no link is supplied, please add this manually to `content/packs/$PACK/Author_image.png`

</div>
</details>

<br/>

<details><summary><strong>Adopt Using Visual Studio Code</strong></summary>
<br/>
<div>

> If you are creating the Pull Request from Visual Studio Code, please follow the below steps. For additional guidance, watch the <a href="https://www.youtube.com/watch?v=9GPkhtRw4Oc">Visual Studio video overview</a>.

Author: I thought what we talked about was having a parent section with two subsections, first one should be the script, second VSCode. 

I think the VSCode video isn't needed unless someone decides they want to do the changes locally and but using the IDE.

    Note: Support for this Pack will be moved to the Partner on MONTH, DAY, YEAR.
    ```
    Make sure you change the `MONTH`, `DAY`, and `YEAR` to the appropriate date that is 90 days from your submission date.

2. Once complete, save these changes and run `demisto-sdk update-release-notes -i <path to pack> -f` to update the release notes. See [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/update_release_notes/README.md). After the command has been completed, it will create the new release note Markdown file in the `ReleaseNotes` folder and update the version number in `pack_metadata.json`. Before continuing, you need to add the following text to the release note: 

```
Start of adoption process.
```

3. Now, it's time to save and commit the changes as a GitHub pull request. Once you publish the changes via Visual Studio, GitHub will prompt you to open a pull request. When prompted, click the green button "Compare & pull request".
    - Double check the pull request to ensure all changes are correct.
    - Change the pull request title to 'Company Name Pack Adoption' and adjust the description to _Updating README file for adoption_.

>     - When ready, click the green button “Create pull request" on the bottom of the page.

Reviewer: ```suggestion
    - When ready, click the green button `Create pull request` on the bottom of the page.
```

4. The request will now be reviewed, approved and merged by a Cortex XSOAR engineer!

**After the 90 days**, another pull request must be submitted to complete the adoption process. Please follow the below steps if you are submitting the final pull request via Visual Studio:
1. Update the release note just as you did in the first pull request but change the text to the below:
    ```
    Note: Support for this Pack moved to the partner on MONTH, DAY, YEAR.

    Please contact the partner directly via the support link on the right.
    ```
2. Next, go to the pack_metadata.json file and update the following sections:

>     - `currentVersion` - update the version. For this example, we would be updating it to 1.2.12

Author: It's referring to the video tutorial where they update the Pack version in `pack_metadata.json`.

1. Go to the `Packs` folder and find your company’s pack.
2. Find the `README.md` file and then click the ![Pencil_Icon](/doc_imgs/partners/Pencil_Icon.png) on the right side of the screen to edit the file. 
3. In the first line of the file, copy and paste the below text to show that the support is moving over to the partner: 
    
    ```
    Note: Support for this Pack will be moved to the Partner on MONTH, DAY, YEAR.
    ```
    
    
    Make sure you change the `MONTH`, `DAY`, and `YEAR` to the appropriate date that is 90 days from your submission date.

4. Once complete, save these changes and run `demisto-sdk update-release-notes -i <path to pack> -f` to update the release notes. See [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/update_release_notes/README.md). After the command has been completed, it will create the new release note Markdown file in the `ReleaseNotes` folder and update the version number in `pack_metadata.json`. Before continuing, you need to add the following text to the release note: 

        
        Start of adoption process.
        

> 

Author: Nice catch! 👀

> 5. Edit the Pull Request title to '`COMPANY_NAME` Pack Adoption' and adjust the description to 'Updating README file for adoption'.

Reviewer: ```suggestion
5. At the bottom of the screen, edit the Pull Request title to '`COMPANY_NAME` Pack Adoption' and adjust the description to 'Updating README file for adoption'.
```

> 6. Create a new branch named `partners-COMPANY_NAME-adoption-start`. 

Reviewer: ```suggestion
6. Create a new branch named `partner-COMPANY_NAME-adoption-start`. 
```

![pack example cyren](/doc_imgs/partners/packexample_cyren.png)

## Process
Choose one of the options below and follow the step by step instructions or watch the video tutorials below to learn how to submit your adoption Pull Requests. 

> <details><summary><strong>Working on a local clone of Content repo</strong></summary>

Reviewer: The nested summary does not look so good as it is not clear what belongs to what and ther's no indentation between sections (see attached gif). 
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/44666568/170583902-b738cd0e-e805-41f6-95bb-ca7f2ce3cffe.gif)

Let's Instead change this to a simple `###` header, like this:
```suggestion
### Option 1: Work on a local clone of the Content repo
These methods assume that you have already [forked the Content repository](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-2-fork-the-github-repo) and [cloned the fork onto your local machine](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-3-clone-the-github-fork-locally).
Choose either one of the methods below to start or complete adoption.

```
And also add an additional header before the [Adopt Using GitHub UI](https://github.com/demisto/content-docs/pull/1103/files#diff-d12af46ecf104a4963eef8a8a31a022f0e69642d0816056c90032653ce98a813R202), such that finally we'll have two headers under the first there will be the `Adopt Using Helper Script` and `Adopt Using a Text Editor/IDE` and under the second there will be the `Adopt Using GitHub UI`

Author: Yea, I tried to do nested section but seems that docusaurus fails when using `blockquotes` tag to create a tabbed sub-section. I then used [their documentation ](https://docusaurus.io/docs/markdown-features#details) but as you can see, the nesting doesn't actually render as expected.

Will use h3 sections instead as suggested.

<br/>

Please follow the below steps if you are working on a cloned GitHub repository from an IDE: 

1. Locate your company's pack folder and open the `README.md` file. Paste the below text into the file: 

    ```
    Note: Support for this Pack will be moved to the Partner on MONTH, DAY, YEAR.
    ```
    Make sure you change the `MONTH`, `DAY`, and `YEAR` to the appropriate date that is 90 days from your submittal date.

>     - Once complete, save these changes and run `demisto-sdk update-release-notes -i <path to pack> -f` to update the release notes. See [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/update_release_notes/README.md). After the command has been completed, it will create the new release note Markdown file in the `ReleaseNotes` folder and update the version number in `pack_metadata.json`. Before continuing, you need to add the following text to the release note: `Start of adoption process, update to readme file`

Reviewer: ```suggestion
    - Once complete, save these changes and run `demisto-sdk update-release-notes -i <path to pack> -f` to update the release notes. See [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/update_release_notes/README.md). After the command has been completed, it will create the new release note Markdown file in the `ReleaseNotes` folder and update the version number in `pack_metadata.json`. Before continuing, you need to add the following text to the release note: `Start of adoption process, visit the pack readme for more info`
```

Reviewer: Let's also fix this ☝️  in the Github UI flow

</details>

<br/>
<strong>Command Line Instructions</strong>
<details>
<summary>Click here to see the command line instructions</summary>

> From within the `content` directory, run:
> ```bash
> ./Utils/adopt_pack.bash
> ```
> 

Reviewer: ```suggestion
This script will automatically perform the necessary steps to create an adoption PR.

Requirements:
Before using this automation, make sure you have [Git](https://git-scm.com/downloads) and [Python](https://www.python.org/downloads/) installed.

Note:
This script will install `demisto-sdk` python package if it does not exist in your environment.

Usage:
1. After [Forking](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-2-fork-the-github-repo) and [Cloning](https://xsoar.pan.dev/docs/tutorials/tut-setup-dev#step-3-clone-the-github-fork-locally) the Content repo to your local environment, you will need to search for your pack name. Go to the `Packs` folder inside the cloned repo, and search for your pack folder, the name of the directory is the pack name that we will use to run the automation with.
2. From within the `content` directory, run:
\```bash
./Utils/adopt_pack.bash "start" "MyPackName"
\```
3. Visit the link given in the output of the script to publish the pull request containing the adoption details created by the automation.  

```
